tests/other/test_fd_pipe.c

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <string.h>
+
+int pipefd[2];
+char buf[5];
+
+int main() {
+  const char *filename = "test.dat";
+
+  // Create a pipe
+  int status = pipe(pipefd);
+  if (status == -1) {
```

Comment:
You can just use assert() here and elsewhere in this file rather then returning codes.   That way when it fails the test will show exactly which line it was that failed.